dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/AbstractImageRegistry.java

```diff
@@ -935,11 +930,7 @@ private List<SourceFile> loadFiles(SourceCodeRepoInterface sourceCodeRepo, Tool
     }
 
     private SourceFile createSourceFile(String path, DescriptorLanguage.FileType type) {
-        SourceFile sourcefile = new SourceFile();
```

Comment:
This seems like a downgrade in readability. If the desire is to convert to a one-liner, we sometimes use a builder pattern so that we can still tell which argument is setting what value.

Reply:
The `create` method exists for the following reasons:
1) It centralizes the creation logic, so we only need to change it in one place.
2) It enforces mandatory arguments at compile time, and we can easily change what's mandatory by modifying the method.  If we use a builder, it's easy to forget to set something.  Unless you use something like a "step builder", which is kinda weird and a lot of code.
Yes, the drawback is that Java doesn't have named arguments, so it's not quite as easy to read.
The goal is to avoid going on an Easter Egg hunt every time we want to make a small change to how SourceFiles are created, to not have to modify the code in 8 different places, and to greatly reduce the chance that we'll forget to change something somewhere and introduce a bug.

Reply:
> I think the function call does improve the overall readability of the code it's embedded in, though, because it takes up less visual and cognitive space (you don't have to decode why all of the setters are there).

The problem is I've been burned by mixing up arguments too many times, I do agree that builders have a trade-off in that you need to validate and throw exceptions at runtime, but given that we need to test this either way, I still find the improvement in readability worthwhile. 

https://blogs.oracle.com/javamagazine/post/exploring-joshua-blochs-builder-design-pattern-in-java (with validation)